yb-voyager/cmd/analyzeSchema.go

```diff
@@ -452,6 +454,25 @@ func checkDDL(sqlInfoArr []sqlInfo, fpath string) {
 			reportCase(fpath, "LANGUAGE C not supported yet.",
 				"", "", "FUNCTION", tbl[2], sqlInfo.formattedStmt)
 			summaryMap["FUNCTION"].invalidCount++
+		} else if regMatch := partitionColumnsRegex.FindStringSubmatch(sqlInfo.stmt); regMatch !=nil {
+			allColumns := strings.Trim(regMatch[3], `()`)
+			allColumns = strings.Trim(allColumns, " ")
+			allColumnsList := strings.Split(allColumns, "(")
+
+			primaryKeyColumns := allColumnsList[len(allColumnsList)-1]
+			primaryKeyColumns = strings.Trim(primaryKeyColumns, `()`)
+			partitionColumns := strings.Trim(regMatch[4], `()`)
+
+			partitionColumnsList := utils.CsvStringToSlice(partitionColumns)
+			primaryKeyColumnsList := utils.CsvStringToSlice(primaryKeyColumns)
+			for _, eachPartitionColumn := range partitionColumnsList {
+				eachPartitionColumn = strings.Trim(eachPartitionColumn, " ")
+				idxInPrimaryKeyColumns := sort.SearchStrings(primaryKeyColumnsList, eachPartitionColumn)
+				if idxInPrimaryKeyColumns == len(primaryKeyColumnsList) || primaryKeyColumnsList[idxInPrimaryKeyColumns] != eachPartitionColumn {
```

Comment:
I don't think `primaryKeyColumnsList[idxInPrimaryKeyColumns]` would return the expected value seeing that we never explicitly sorted this slice..

Reply:
Oh yes, thanks. I assumed that this sort.SearchStrings function will sort the Slice and then search.